core/src/main/scala/kafka/server/BrokerMetadataCheckpoint.scala

```diff
@@ -38,6 +38,7 @@ class BrokerMetadataCheckpoint(val file: File) extends Logging {
         val brokerMetaProps = new Properties()
         brokerMetaProps.setProperty("version", 0.toString)
         brokerMetaProps.setProperty("broker.id", brokerMetadata.brokerId.toString)
+        brokerMetaProps.setProperty("cluster.id", brokerMetadata.clusterId)
```

Comment:
When writing new checkpoint files, do we require the clusterID to be defined?

Reply:
@hachikuji Yes, `clusterId` is always required and checkpointed. Do you see a case where it shouldn't be? My understanding is that `clusterId` is always defined thus it makes sense to always checkpoint it.

Reply:
I think it should always be present. My question was mostly whether it's worth having a null check.

Reply:
I think it is not necessary as `clusterId` is always defined.